examples/gpt/tools.py

```diff
@@ -263,24 +260,39 @@ def add_dict(config, k, v):
         for ik, iv in v.items():
             add_dict(config, ik, iv)
 
-    for key, value in yaml_dict.items():
-        add_dict(args, key, value)
-
     print(
         '------------------------ arguments ------------------------',
         flush=True)
-    str_list = []
-    for key, value in args.items():
-        dots = '.' * (48 - len(key))
-        str_list.append('  {} {} {}'.format(key, dots, value))
-    for arg in sorted(str_list, key=lambda x: x.lower()):
-        print(arg, flush=True)
+
+    for key, value in yaml_dict.items():
+        args = {}
+        add_dict(args, key, value)
+
+        print("{}:".format(key), flush=True)
+        str_list = []
+        for key, value in args.items():
+            dots = '.' * (48 - len(key))
+            str_list.append('  {} {} {}'.format(key, dots, value))
+        for arg in sorted(str_list, key=lambda x: x.lower()):
+            print(arg, flush=True)
+
     print(
         '-------------------- end of arguments ---------------------',
         flush=True)
 
 
-def parse_yaml_auto(yaml_args):
+def parse_yaml_auto():
```

Comment:
Move `yaml parser`  funciton to another file, such as yaml.py or yaml_config.py
Don't set everything of tools into only one file.